rust/lance-io/src/lib.rs

```diff
@@ -200,6 +200,18 @@ impl ReadBatchParams {
             )),
         }
     }
+
+    pub fn to_offsets_total(&self, total: u32) -> Result<PrimitiveArray<UInt32Type>> {
+        match self {
+            Self::Indices(indices) => Ok(indices.clone()),
+            Self::Range(r) => Ok(UInt32Array::from(Vec::from_iter(
+                r.start as u32..r.end as u32,
+            ))),
```

Comment:
`from_iter_values()` is what you are looking for:
```suggestion
            Self::Range(r) => Ok(UInt32Array::from_iter_values(
                r.start as u32..r.end as u32,
            )),
```